src/test/java/spoon/test/factory/MethodFactoryTest.java

```diff
@@ -31,4 +32,30 @@ public void testCreateReference() {
 
         assertThat(actualCreatedReference, is(expectedReference));
     }
-}
+
+    @Test
+    public void testCreateReferenceWithActualMethod() throws ClassNotFoundException, NoSuchMethodException {
+        // contract: createReference creates a method reference of a actual method foo
+
+        // arrange
+        Factory factory = new Launcher().getFactory();
+        Class<?> testClass = Class.forName("spoon.test.factory.testclasses.Bar");
+        Method testMethod = testClass.getMethod("foo");
+
+        CtExecutableReference<?> expectedReference = factory.createExecutableReference();
+        expectedReference.setSimpleName("foo");
+        CtTypeReference<?> ctTypeReference = factory.Type().createReference(Bar.class);
+        expectedReference.setDeclaringType(ctTypeReference);
+        CtTypeReference<?> typeReference = factory.Type().createReference(testMethod.getReturnType());
+//        expectedReference.setType(typeReference);
```

Comment:
I am not able to figure out how to set the type of the expected reference to "void", I need help here

Reply:
You mean something like this?

```java
expectedReference.setType(factory.Type().voidPrimitiveType());
```

Reply:
yes I meant something like this only, thank you so much